x/stakeibc/keeper/unbonding_records.go

```diff
@@ -43,171 +65,250 @@ func (k Keeper) CreateEpochUnbondingRecord(ctx sdk.Context, epochNumber uint64)
+// As a result, unbondings lead to a more balanced distribution of stake across validators
+//
+// Context:
+//   Over time, as LSM Liquid stakes are accepted, the total stake managed by the protocol becomes unbalanced
+//   as liquid stakes are not aligned with the validator weights. This is only rebalanced once per unbonding period
+func (k Keeper) UnbondFromHostZone(ctx sdk.Context, hostZone types.HostZone) error {
+	k.Logger(ctx).Info(utils.LogWithHostZone(hostZone.ChainId,
+		"Preparing MsgUndelegates from the delegation account to each validator"))
+
+	// Confirm the delegation account was registered
+	if hostZone.DelegationIcaAddress == "" {
+		return errorsmod.Wrapf(types.ErrICAAccountNotFound, "no delegation account found for %s", hostZone.ChainId)
+	}
+
+	// Iterate through every unbonding record and sum the total amount to unbond for the given host zone
+	totalAmountToUnbond, epochUnbondingRecordIds := k.GetTotalUnbondAmountAndRecordsIds(ctx, hostZone.ChainId)
+	k.Logger(ctx).Info(utils.LogWithHostZone(hostZone.ChainId,
+		"Total unbonded amount: %v%s", totalAmountToUnbond, hostZone.HostDenom))
+
+	// If there's nothing to unbond, return and move on to the next host zone
+	if totalAmountToUnbond.IsZero() {
+		return nil
+	}
+
+	// Determine the ideal balanced delegation for each validator after the unbonding
+	//   (as if we were to unbond and then rebalance)
+	// This will serve as the starting point for determining how much to unbond each validator
+	currentTotalDelegation := k.GetTotalValidatorDelegations(hostZone)
+	delegationAfterUnbonding := currentTotalDelegation.Sub(totalAmountToUnbond)
+	balancedDelegationsAfterUnbonding, err := k.GetTargetValAmtsForHostZone(ctx, hostZone, delegationAfterUnbonding)
+	if err != nil {
+		return errorsmod.Wrapf(err, "unable to get target val amounts for host zone %s", hostZone.ChainId)
+	}
+
+	// Determine the unbond capacity for each validator
+	// Each validator can only unbond up to the difference between their current delegation and their balanced delegation
+	// The validator's current delegation will be above their balanced delegation if they've received LSM Liquid Stakes
+	//   (which is only rebalanced once per unbonding period)
+	validatorUnbondCapacity := k.GetValidatorUnbondCapacity(ctx, hostZone.Validators, balancedDelegationsAfterUnbonding)
+	if len(validatorUnbondCapacity) == 0 {
+		return fmt.Errorf("there are no validators on %s with sufficient unbond capacity", hostZone.ChainId)
+	}
+
+	// Sort the unbonding capacity by priority
+	// Priority is determined by checking the how proportionally unbalanced each validator is
+	// Zero weight validators will come first in the list
+	prioritizedUnbondCapacity := SortUnbondingCapacityByPriority(validatorUnbondCapacity)
+
+	// Get the undelegation ICA messages and split delegations for the callback
+	msgs, unbondings, err := k.GetUnbondingICAMessages(hostZone, totalAmountToUnbond, prioritizedUnbondCapacity)
```

Comment:
what's the max number of unbonding msgs we can fix in the host zone ICA block size? recall the limit is ~35 for delegation ICA msgs. We should probably enforce a cap on the # of msgs here as well for now, maybe implemented inside GetUnbondingICAMessages based on a param

Reply:
ah yeah good call! Wdyt about just adding the size as a constant rather than a param? Feels like we've been going kinda crazy with the module params and none of them are actually used

Reply:
on board with that

Reply:
It shouldn't be a param anyway, I thought the purpose of the param was something that could be voted on to change dynamically -- it should be something which the stride users have some say in because it affects them to be a param, anything which is a technical tuning should just be done internally by us in development.